LeetCodeSolutions/SpellsAndPotions/SpellsAndPotions.cs

```diff
@@ -0,0 +1,72 @@
+namespace LeetCodeSolutions;
+public class SpellsAndPotionsSolution
+{
+
+
+    private class PotionsContainer
+    {
+        private readonly int[] potions;
+        private readonly long success;
+        public PotionsContainer(int[] potions, long success)
+        {
+            this.potions = potions;
+            Array.Sort(this.potions);
+            this.success = success;
+        }
+
+        private int BinarySearch(decimal target, int i, int j)
+        {
+            while (i <= j)
+            {
+                if (i == j)
```

Comment:
try moving this evaluation down to the return statement, make for a clean return in a ternary and can just evauate while (i<=j)?